core/src/main/scala/org/apache/spark/h2o/H2OContext.scala

```diff
@@ -218,9 +218,14 @@ class H2OContext private (val sparkContext: SparkContext, conf: H2OConf) extends
     * @param stopSparkContext  stop also spark context
     */
   def stop(stopSparkContext: Boolean = false): Unit = {
-    announcementService.shutdown
-    backend.stop(stopSparkContext)
-    H2OContext.stop(this)
+    if(!stopped) {
```

Comment:
This can be in theory invoked by multiple threads -  it should be synchronized on h2oContext.